src/main/java/com/snowflake/kafka/connector/records/RecordService.java

```diff
@@ -177,7 +190,20 @@ public Map<String, Object> getProcessedRecordForStreamingIngest(SinkRecord recor
     final Map<String, Object> streamingIngestRow = new HashMap<>();
     for (JsonNode node : row.content.getData()) {
       try {
-        streamingIngestRow.put(TABLE_COLUMN_CONTENT, MAPPER.writeValueAsString(node));
+        if (schematizationEnable) {
```

Comment:
Why we still need to convert the SinkRecord to a SnowflakeTableRow first and then to a MAP object, could we convert the SinkRecord to MAP directly?

Reply:
I see, would it be hard to extract the error detection logic out as a helper function and use it in both cases?